aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/PersistentMutationOutbox.java

```diff
@@ -414,5 +425,25 @@ private Completable unexpectedMutationScenario() {
                 "Please report at https://github.com/aws-amplify/amplify-android/issues."
             ));
         }
+
+        private PendingMutation<Model> merge(PendingMutation<T> incoming, PendingMutation<T> existing)
+                throws AmplifyException {
+            SerializedModel incomingSerializedModel = SerializedModel.create(incoming.getMutatedItem(),
```

Comment:
Instead of creating a new `SerializedModel` instance (e.g. `SerializedModel.create(...)`, we should just use the object itself, if  `incoming.getMutatedItem() instanceof SerializedModel`.    If it's not, fallback to existing behavior (overwrite instead of merge)